### PR TITLE
chore:👷‍♀️Sets up dev environment and CI checks

### PR DESCRIPTION
**Pull request by ekwoka**
This PR:
- Adjusts settings and configs for linting/formatting/testing
- Adds precommit hook for linting and typechecking
- Adds workflows to check PRs

### CODE REVIEW

**Comment by ekwoka** [comment]
Why do we need the word "chore" when we have 👷‍♀️ ?

👷‍♀️ = Updates to CI/CD

**Comment by wiktoriavh** [comment]
@ekwoka We use conventional commits here.

File: builder-integration.ts [hunk 1]
```diff
@@ -751,9 +789,11 @@ export function builderDevTools(opts: BuilderioOptions = {}): Plugin {
         req.socket.address();
 
         const orgResponseEnd = res.end;
-        res.end = function (...args: any[]) {
+        res.end = function (...args: unknown[]) {
```
Comment (wiktoriavh) [inline builder-integration.ts:792]: question: Why are you not typing args the same way as you did on line 811?

**Review by wiktoriavh — APPROVED** [review]

File: builder-integration.ts (hunk 1 above)
Comment (ekwoka) [reply builder-integration.ts:792]: Because, I didn't write this file.

**Comment by ekwoka** [comment]
But what convention?

**Pull request merged**